pkg/reduce/pbq/store/wal/segment.go

```diff
@@ -225,10 +224,8 @@ func (w *WAL) encodeReadMessageHeader(message *isb.ReadMessage, messageLen int64
 	return buf, nil
 }
 
-func (w *WAL) encodeReadMessageBody(message *isb.ReadMessage) (*bytes.Buffer, error) {
-	m := new(bytes.Buffer)
-	enc := gob.NewEncoder(m)
-	err := enc.Encode(message.Message)
+func (w *WAL) encodeReadMessageBody(readMsg *isb.ReadMessage) ([]byte, error) {
+	msgBinary, err := readMsg.Message.MarshalBinary()
```

Comment:
This function is named `encodeReadMessageBody` but it's actually encoding `ReadMessage.Message` field where
```go
type Message struct {
	Header
	Body
}
```

Reply:
fix the function name?

Reply:
Do not need to wrap it as a function anymore maybe.